app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/amend/AmendReturnController.scala

```diff
@@ -124,17 +124,35 @@ class AmendReturnController @Inject() (
   def checkYourAnswers(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
       withStartingToAmendReturn(request) { journey =>
-        Ok(
-          checkYourAnswersPage(
-            journey.originalReturn.completeReturn,
-            rebasingEligibilityUtil,
-            journey.subscribedDetails,
-            journey.originalReturn.completeReturn.representativeType(),
-            journey.originalReturn.completeReturn.isIndirectDisposal(),
-            Some(journey.originalReturn.returnType.isFurtherOrAmendReturn),
-            controllers.returns.routes.ViewReturnController.displayReturn()
-          )
-        )
+        val isSAquestionAnswered: Option[Boolean] = journey.originalReturn.completeReturn.triageAnswers
+          .fold(_.alreadySentSelfAssessment, c => c.alreadySentSelfAssessment)
+        val originalTaxYear                       = journey.originalReturn.summary.taxYear
+        val currentTaxYear                        = TaxYear.thisTaxYearStartDate().getYear.toString
+        val areFutureDatesEnabled                 = viewConfig.enableFutureDates
+        val isSubmissionInPreviousTaxYear         = originalTaxYear =!= currentTaxYear
+
+        (isSAquestionAnswered.isEmpty, areFutureDatesEnabled, isSubmissionInPreviousTaxYear) match {
+          case (true, false, true) =>
+            Redirect(
+              controllers.returns.triage.routes.CommonTriageQuestionsController.haveYouAlreadySentSelfAssessment()
+            )
+          case (true, true, _)     =>
+            Redirect(
+              controllers.returns.triage.routes.CommonTriageQuestionsController.haveYouAlreadySentSelfAssessment()
+            )
```

Comment:
This block code is bit confusion. I feel it has some unnecessary conditions. Please refine it more better

`        (isSAquestionAnswered.isEmpty, areFutureDatesEnabled, isSubmissionInPreviousTaxYear) match {
          case (true, false, true) =>
            Redirect(
              controllers.returns.triage.routes.CommonTriageQuestionsController.haveYouAlreadySentSelfAssessment()
            )
          case (true, true, _)     =>
            Redirect(
              controllers.returns.triage.routes.CommonTriageQuestionsController.haveYouAlreadySentSelfAssessment()
            )`